zerver/lib/upload.py

```diff
@@ -788,6 +795,14 @@ def upload_export_tarball(self, realm: Realm, tarball_path: str) -> str:
         public_url = realm.uri + '/user_avatars/' + path
         return public_url
 
+    def delete_export_tarball(self, uri: str) -> Optional[str]:
+        # Using `urllib` here would require some extra string manipulation to
+        # remove '/user_avatars/' from the path.
+        path = re.sub('http://zulip.testserver/user_avatars/', '', uri)
```

Comment:
Reading this, I think we should just store the (relative URL) `path` part, i.e. `/user_avatars/...` in the `extra_data` field for these; this code is going to be broken in the case that the server's URL changes.  And when rendered in a browser as a link, the right thing will happen of automatically inserting the server's URL.

Reply:
(It also lets us get rid of this weird `re.sub` line)

Reply:
Yeah, it's clear as day now how broken this was. Updated.